### PR TITLE
refactor(datatype): remove `Category` type and related APIs

### PR DESCRIPTION
**Pull request by cpcloud**
This PR removes the Category datatype and associated APIs. The `label` API, which was the only user facing API now exists on `IntegerValue`s.

### CODE REVIEW

**Comment by jreback** [comment]
is this replaced by anything?

**Comment by cpcloud** [comment]
@jreback Yep! It's replaced by `IntegerColumn`.

`bucket` remains as it was on `NumericColumn` and `label` has been moved to a method on `IntegerColumn` instead of a specific operation that only worked on categories.

**Comment by jreback** [comment]
ok thanks @cpcloud

**Comment by kszucs** [comment]
I wanted to remove Category dtype as well, nice cleanup!

File: ibis/expr/operations/histograms.py [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
     close_extreme = rlz.optional(rlz.instance_of(bool), default=True)
     include_under = rlz.optional(rlz.instance_of(bool), default=False)
     include_over = rlz.optional(rlz.instance_of(bool), default=False)
     output_shape = rlz.Shape.COLUMNAR
 
-    @property
+    @attribute.default
     def output_dtype(self):
-        return dt.Category(self.nbuckets)
+        import ibis.expr.datatypes as dt
```
Comment (kszucs) [inline ibis/expr/operations/histograms.py:24]: Can you import it at the top?

**Review by kszucs — APPROVED** [review]

**Pull request merged**